magma/protocol_type.py

```diff
@@ -87,9 +125,20 @@ def wire(self, other):
     def __imatmul__(self, other):
         self.wire(other)
 
+    def __getitem__(self, idx):
```

Comment:
feel pretty strongly about not having these defined on protocols. see above comment. furthermore, it means we can't define these custom functions on the protocols.

Reply:
deleted